### ISSUE TITLE
Dockerfile variable expansion syntax not parsed correctly

### ISSUE THREAD

**Issue by sanmai-NL** [hadolint/hadolint#994]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

A feature of Dockerfile syntax (1.6.0) is [shell parameter expansion](https://github.com/containers/buildah/issues/4284). In Docker Engine, Podman and nerdctl, these can be used to, e.g., fail a build when a build arg is unset. I haven't found documentation for the feature by Docker Inc.

### Expected behavior

Shell parameter expansions are parsed. For Hadolint, I presume this is no more than expanding the grammar/parser, without doing anything with the semantics.

### Actual behavior

Shell parameter expansions not parsed, resulting in fatal failure.

### Steps to reproduce the behavior

```console
$ hadolint Containerfile

Containerfile:5:31 unexpected ':' expecting '@', '\', a new line followed by the next instruction, or the image tag
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```text
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker.io/docker/dockerfile:1@sha256:ac85f380a63b13dfcefa89046420e1781752bab202122f8f50032edf31be0021
ARG IMAGE_PREFIX=
ARG DISTRO=alpine
ARG VERSION_DISTRO=3.19
FROM ${IMAGE_PREFIX:?}${DISTRO:?}:${VERSION_DISTRO:?}
```

**Additional environment details (OS, stack version, etc.)**
